### PR TITLE
Improve Sonos notification timeout handling

### PR DESCRIPTION
**Pull request by ivivanov-bg**
1. Extract the notification timeout as a thing configuration parameter
2. Increase the default notification timeout to 40 sec. (as 20 s is too short time)

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.sonos/ESH-INF/config/config.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+			<description>Specifies the amount of time for which the notification sound will be played</description>
+			<default>40</default>
+		</parameter>
+		<parameter name="refresh" type="integer">
+			<label>Refresh interval</label>
+			<description>Specifies the refresh interval in seconds</description>
+			<default>60</default>
+		</parameter>
+	</config-description>
+</config-description:config-descriptions>
```
Comment (hakan42) [inline extensions/binding/org.eclipse.smarthome.binding.sonos/ESH-INF/config/config.xml]: Please add a linefeed at the end of this file.

File: extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java [hunk 2]
```diff
@@ -205,8 +211,16 @@ public void initialize() {
...
-            updateStatus(ThingStatus.ONLINE);
             onUpdate();
+            
+            if (getConfigAs(ZonePlayerConfiguration.class).notificationTimeout == null) {
+                Configuration c = editConfiguration();
+                c.put(ZonePlayerConfiguration.NOTIFICATION_TIMEOUT, DEFAULT_NOTIFICATION_TIMEOUT);
+                updateConfiguration(c);
+            }
+            
+            this.notificationTimeout = getConfigAs(ZonePlayerConfiguration.class).notificationTimeout;
```
Comment (hakan42) [inline extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java]: Please initiliaze the notification Timeout to the default here and don't use the NPE-safety check in the getter?

File: extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java [hunk 3]
```diff
@@ -2801,4 +2815,8 @@ private long sleepStrTimeToSeconds(String sleepTime) {
         int seconds = Integer.parseInt(units[2]);
         return 3600 * hours + 60 * minutes + seconds;
     }
+    
+    protected Integer getNotificationTimeout() {
+        return this.notificationTimeout != null ? this.notificationTimeout : DEFAULT_NOTIFICATION_TIMEOUT;
```
Comment (hakan42) [inline extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java]: This safety check is not necessary (and the getter a standard getter method) if you initialize the default in the initialize() method (see my previous comment)

File: extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java (hunk 2 above)
Comment (ivivanov-bg) [inline extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java]: done

File: extensions/binding/org.eclipse.smarthome.binding.sonos/ESH-INF/config/config.xml (hunk 1 above)
Comment (ivivanov-bg) [inline extensions/binding/org.eclipse.smarthome.binding.sonos/ESH-INF/config/config.xml]: done

**Comment by sjka** [comment]
Thank you both!

**Pull request merged**